otx/mpa/modules/hooks/eval_before_train_hook.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from mmcv.runner import HOOKS
+from mmcv.runner import EvalHook as SegEvalHook
+from mmcv.runner import Hook
+
+from otx.mpa.modules.hooks.checkpoint_hook import CheckpointHookWithValResults
+from otx.mpa.modules.hooks.eval_hook import CustomEvalHook as ClsEvalHook
+
+try:
+    from mmdet.core.evaluation.eval_hooks import EvalHook as DetEvalHook
+except ImportError:
+    DetEvalHook = None
+
+
+@HOOKS.register_module()
+class EvalBeforeTrainHook(Hook):
+    """Hook to evaluate and save model weight before training."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._executed = False
+
+    def before_train_epoch(self, runner):
```

Comment:
I see you execute the existing hook methods dynamically once before training. Isn't it okay to move this before_run()? And I just wonder whether it's gonna be more complicated if we do not make use of existing hook and write codes for it. Calling after_train_epoch of hook in before_train_epoch is hard to understand workflow.

Reply:
As far as I know, some function modifies the model in `before_train_run()`. So I use `before_train_epoch()`.
But I'll check that whether I can use `beore_run()`.
and for second question, the reason why I use exiting hook is that I want the initial model weight can be chosen if initial weight is best model weight. Each validation hook, which I use, validates model score and saves model weight if needed. So, I just want to use this way not to be chosen as best weight naturally.

Reply:
I checked whether before_run() also works well but I found that accuracies of consecutive epochs are exactly same.
I tried to find why it happens, but I didn't able to find a reason.
I think it's better to use before_train_epoch() and change it to before_run() if needed.